policybot/pkg/testflakes/testflakes.go

```diff
@@ -0,0 +1,358 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Take in a pr number from path "istio-prow/pr-logs/pull/istio-istio" and examine the pr
+ * for all tests that are run and their results. The results are then written to Spanner.
+ */
+package testflakes
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+
+	store "istio.io/bots/policybot/pkg/storage"
+)
+
+/*
+ * Pull struct for the structure under refs/pulls in clone-records.json
+ */
+type Pull struct {
+	Number int
+	Author string
+	Sha    string
+}
+
+/*
+ * Cmd struct for Commands object under clone-records.json
+ */
+type Cmnd struct {
+	Command string
+	Output  string
+}
+
+/*
+ * Finished struct to store values for all fields in Finished.json
+ */
+type Finished struct {
+	Timestamp int64
+	Passed    bool
+	Result    string
+}
+
+/*
+ * Clone_Record struct to store values for all fields in clone-records.json
+ */
+type CloneRecord struct {
+	Refs struct {
+		Org       string
+		Repo      string
+		BaseRef   string
+		BaseSha   string
+		Pulls     []Pull
+		PathAlias string
+	}
+	Commands []Cmnd
+	Failed   bool
+}
+
+/*
+ * Tests strut to keep track of the test suite names and the directory for each test runs for the pr.
+ */
+type Tests struct {
+	Name string
+	Prs  []string
+}
+
+/*
+ * Started struct to store values from started.json
+ */
+type Started struct {
+	Timestamp int64
+}
+
+type PrFlakeTest struct {
+	client *storage.Client
+	ctx    context.Context
+}
+
+/*
+ * Function to initionalize new PrFlakeTest object.
+ * Use `prFlakeyTest, err := NewPrFlakeTest()` and `testFlakes := prFlakeyTest.checkTestFlakesForPr(<pr_number>)`
+ * to get test flakes information for a given pr.
+ */
+func NewPrFlakeTest() (*PrFlakeTest, error) {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &PrFlakeTest{
+		client: client,
+		ctx:    ctx,
+	}, nil
+}
+
+/*
+ * Contains function check if a string exists in a given slice of strings.
+ */
+func contains(slic []string, ele string) bool {
+	for _, e := range slic {
+		if strings.Compare(e, ele) == 0 {
+			return true
+		}
+	}
+	return false
+}
+
+/*
+ * GetTest function get all directories under the given pr in istio-prow/pr-logs/pull/istio-istio/PRNUMBER for each test suite name.
+ * Client: client used to get buckets and objects.
+ * PrNum: the PR number inputted.
+ * Return []Tests return a slice of Tests objects.
+ */
+func (prFlakeTest PrFlakeTest) getTests(prNumInt int64) ([]Tests, error) {
+	prNum := strconv.FormatInt(prNumInt, 10)
+	ctx := prFlakeTest.ctx
+	client := prFlakeTest.client
+	bucket := client.Bucket("istio-prow")
```

Comment:
These strings should be coming from configuration rather than being hardcoded. Consume these as inputs to the NewXXX function.